src/common/types/interval.cpp

```diff
@@ -500,6 +500,47 @@ date_t Interval::Add(date_t left, interval_t right) {
 	return result;
 }
 
+bool Interval::TryAdd(date_t left, interval_t right, date_t &result) {
+	if (!Date::IsFinite(left)) {
+		return false;
+	}
+	if (right.months != 0) {
+		int32_t year, month, day;
+		Date::Convert(left, year, month, day);
+		int32_t year_diff = right.months / Interval::MONTHS_PER_YEAR;
+		year += year_diff;
+		month += right.months - year_diff * Interval::MONTHS_PER_YEAR;
```

Comment:
This is adding a lot of duplicate code - in general when we add the `Try` methods, we rewrite the regular `Add` methods to use the `Try`, i.e. `Interval::Add` becomes something like:


```cpp
interval_t result;
if (!Interval::TryAdd(left, right, result)) {
   throw OutOfRangeException("Date out of range");
}
return result;
```

Could we either do that, or revert to extending the `TryAddOperator` for `date_t/int32_t` as before?